clvm-traits/src/lib.rs

```diff
@@ -4,12 +4,16 @@
 //! as well as many other values in the standard library that would be common to encode.
 //!
 //! As well as the built-in implementations, this library exposes two derive macros
-//! for implementing the `ToClvm` and `FromClvm` traits on structs. They be marked
-//! with one of the following encodings:
+//! for implementing the `ToClvm` and `FromClvm` traits on structs and enums. They be
+//! marked with the following representations:
 //!
 //! * `#[clvm(tuple)]` for unterminated lists such as `(A . (B . C))`.
 //! * `#[clvm(list)]` for proper lists such as `(A B C)`, or in other words `(A . (B . (C . ())))`.
 //! * `#[clvm(curry)]` for curried arguments such as `(c (q . A) (c (q . B) (c (q . C) 1)))`.
+//!
+//! Additionally, you can use `#[clvm(untagged)]` on enums which don't have a numeric discriminant value.
```

Comment:
and `untagged` isn't allowed on non-enums, right?
I think you should describe what it means for an enum to not encode or match a discriminant.
You pick one matching enum value, is it the first matching from the top? The best matching? is it an error to have a value that matches more than one enum?

I think almost everything you put in the PR description belongs in the documentation on how to use the `clvm` derive macro.